### PR TITLE
Deduplicate crash handlers

### PR DESCRIPTION
**Pull request by tokatoka**
deduplicate codes from timeout handler, panic_hook, crash handlers

### CODE REVIEW

File: libafl/src/executors/inprocess.rs [hunk 1]
```diff
@@ -527,6 +522,70 @@ pub fn inprocess_get_input<'a, I>() -> Option<&'a I> {
...
 
+use crate::{
+    corpus::{Corpus, Testcase},
+    events::Event,
+    state::HasMetadata,
+};
+
+#[inline]
+#[allow(clippy::too_many_arguments)]
+/// Save sate if it is interesting
```
Comment (domenukk) [inline libafl/src/executors/inprocess.rs:533]: What... does that `interesting` mean here?

File: libafl/src/executors/inprocess.rs [hunk 2]
```diff
@@ -527,6 +522,70 @@ pub fn inprocess_get_input<'a, I>() -> Option<&'a I> {
...
+) where
+    E: HasObservers,
+    EM: EventFirer<State = E::State> + EventRestarter<State = E::State>,
+    OF: Feedback<E::State>,
+    E::State: HasSolutions + HasClientPerfMonitor,
+    Z: HasObjective<Objective = OF, State = E::State>,
+{
+    let observers = executor.observers_mut();
+
+    observers
```
Comment (domenukk) [inline libafl/src/executors/inprocess.rs:550]: The name `save_state_for_restart` doesn't really show that observers would be called, imho we should make this more clear, `run_observers_and_save_state` or something? Or an extra method for running the observers

File: libafl/src/executors/inprocess.rs (hunk 1 above)
Comment (tokatoka) [reply libafl/src/executors/inprocess.rs:533]: i'll change it to save if it is an objective

File: libafl/src/executors/inprocess.rs (hunk 2 above)
Comment (tokatoka) [reply libafl/src/executors/inprocess.rs:550]: ok

**Pull request merged**